### ISSUE TITLE
Option to disable cahing of SERVFAIL responses

### ISSUE THREAD

**Issue by gshpychka** [AdguardTeam/AdGuardHome#7376]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

### The problem

I would like to be able to disable the cahing of SERVFAIL responses by adguard.

### Proposed solution

A configuration option to disable the caching of SERVFAIL requests and/or treat it as a failure

### Alternatives considered and additional information

It has been stated in https://github.com/AdguardTeam/AdGuardHome/issues/5924 that this is intended behavior, but the spec does not require the server to do this. So I'm asking for the ability to disable it.

**Comment by masx200** [comment]
![屏幕截图 2024-10-26 172707](https://github.com/user-attachments/assets/8ac5deac-5c41-4175-9d5d-5f5001c37a4f)

**Comment by gshpychka** [comment]
> ![屏幕截图 2024-10-26 172707](https://private-user-images.githubusercontent.com/34191203/380390684-8ac5deac-5c41-4175-9d5d-5f5001c37a4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzY0ODAsIm5iZiI6MTcyOTk3NjE4MCwicGF0aCI6Ii8zNDE5MTIwMy8zODAzOTA2ODQtOGFjNWRlYWMtNWM0MS00MTc1LTlkNWQtNWY1MDAxYzM3YTRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDIwNTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjYzZiMGRlNTM0NGRlMTcxNWRhMjZhNjNlZTk5YTg3OWZkNGUwZjdiNzY0NDczZjhjMTU5OTE4MDg0OTVkODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.O1gDUXZ7WSbc5IUMzKtdYdOoSdvyLLmXgXWOBhVlxTw)

I'm afraid I don't speak Chinese - what is your screenshot showing?

**Comment by masx200** [comment]
> I'm afraid I don't speak Chinese - what is your screenshot showing?

Because tailscale's dns server gets an error servfail

**Comment by B0G0311** [comment]
Just poking to see if there is any update on this? I just spent 1.5 days fumbling with my childs story audioplayer jus to find out its because of caching these servfail responses. Currently canceled caching and my problems are fixed but my dns queries take much longer. Im surprised this is not a much bigger problem.

**Comment by gshpychka** [comment]
> Just poking to see if there is any update on this? I just spent 1.5 days fumbling with my childs story audioplayer jus to find out its because of caching these servfail responses. Currently canceled caching and my problems are fixed but my dns queries take much longer. Im surprised this is not a much bigger problem.

Yeah, would be good to get any sort of response from the team on this. I believe they use emojis to track interest, so be sure to leave a thumbs up on the issue if you'd like to see this implemented

**Comment by jordansworld** [comment]
dang dood where is the adguard team they usually reply faster to me.

**Comment by jordansworld** [comment]
@Versty

**Comment by hupohupochuan** [comment]
I also encountered this problem. The cache could not be cleared, causing the URL www.amd.com to be inaccessible.

<img width="2560" height="1190" alt="Image" src="https://github.com/user-attachments/assets/2eb7be91-acba-4e20-a56a-70d59b89130c" />

**Comment by HStep20** [comment]
Having just found out my problem Ive had for months of weird internet issues was caused by cache SERVFAIL responses, this would be great. I don't want to disable caching entirely, but thats what Ive done and my weird connection issues are solved
